cache/cache.go

```diff
@@ -78,9 +78,11 @@ func (c *randomEvictedCache) Add(key string, data Measurable) {
 	}
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	if _, ok := c.data[key]; !ok {
-		c.cachedBytes += increase
+	if v, ok := c.data[key]; ok {
+		decrease := 2*len(key) + v.Size()
```

Comment:
It seems like the pattern `2*len(key) + val.Size()` is used a bunch: should it be a function?